src/Servers/Kestrel/Core/src/Middleware/LoggingConnectionMiddleware.cs

```diff
@@ -39,6 +48,58 @@ await using (var loggingDuplexPipe = new LoggingDuplexPipe(context.Transport, _l
             }
         }
 
+        public async Task OnConnectionAsync(MultiplexedConnectionContext context)
+        {
+            Debug.Assert(_multiplexedNext != null);
+
+            await _multiplexedNext(new LoggingMultiplexedConnectionContext(context, _logger));
+        }
+
+        private class LoggingMultiplexedConnectionContext : MultiplexedConnectionContext
+        {
+            private readonly MultiplexedConnectionContext _inner;
+            private readonly ILogger _logger;
+
+            public LoggingMultiplexedConnectionContext(MultiplexedConnectionContext inner, ILogger logger)
+            {
+                _inner = inner;
+                _logger = logger;
+            }
+
+            public override string ConnectionId { get => _inner.ConnectionId; set => _inner.ConnectionId = value; }
+            public override IFeatureCollection Features => _inner.Features;
+            public override IDictionary<object, object?> Items { get => _inner.Items; set => _inner.Items = value; }
+
+            public override void Abort()
+            {
+                _inner.Abort();
+            }
+
+            public override void Abort(ConnectionAbortedException abortReason)
+            {
+                _inner.Abort(abortReason);
+            }
+
+            public override async ValueTask<ConnectionContext?> AcceptAsync(CancellationToken cancellationToken = default)
+            {
+                var context = await _inner.AcceptAsync(cancellationToken);
+                if (context != null)
+                {
+                    context.Transport = new LoggingDuplexPipe(context.Transport, _logger);
```

Comment:
Unlike regular connection logging, this is never disposed.

Reply:
Maybe we should decorate the returned ConnectionContext to intercept Dispose().

Reply:
ConnectionContext doesn't have DisposeAsync.

Reply:
It does: https://github.com/dotnet/aspnetcore/blob/ace9b758fdd6ba0e8ff673b7212cb84e3ef1615e/src/Servers/Connections.Abstractions/src/BaseConnectionContext.cs#L16

Reply:
Opps. I only looked at abstract methods. I'll try that.